plugins/reporters/fossid/src/main/kotlin/FossIdReporter.kt

```diff
@@ -93,18 +93,23 @@ class FossIdReporter : Reporter {
         val service = FossIdRestService.create(serverUrl)
 
         return runBlocking(Dispatchers.IO) {
-            input.ortResult.getScanResults().values.flatten().mapNotNull { result ->
-                result.additionalData[SCAN_CODE_KEY]?.let { scanCode ->
-                    async {
-                        logger.info { "Generating report for scan $scanCode." }
-                        service.generateReport(user, apiKey, scanCode, reportType, selectionType, outputDir)
-                            .onFailure {
-                                it.showStackTrace()
-                                logger.info {
-                                    "Error during report generation: ${it.collectMessages()}."
-                                }
-                            }.getOrNull()
-                    }
+            val scanResults = input.ortResult.getScanResults().values.asSequence().flatten()
```

Comment:
BTW, what's the benefit of the `asSequence()` call? As always all values are processed, there's no savings in computations, or?

Reply:
In my version with several calls in the chain, it was a recommendation from Intellij.
I removed in your version because as you said it is not needed.